### PR TITLE
New policy build workflow

### PR DESCRIPTION
**Pull request by Springstone**
This pull request introduces several significant changes to the PowerShell scripts and GitHub workflows to improve policy management and testing. The key changes include the addition of new functions for JSON handling in PowerShell, new unit tests for policy builds, a new workflow for checking policy builds, and the removal of an outdated workflow.

### New Functions for JSON Handling:
* Added `Remove-JSONMetadata` function to remove metadata from JSON objects and handle nested templates. (`.github/actions-pester/PolicyPesterTestHelper.psm1`)
* Added `ConvertTo-OrderedHashtable` function to convert JSON strings to ordered hashtables, ensuring consistent key order. (`.github/actions-pester/PolicyPesterTestHelper.psm1`)

### New Unit Tests:
* Introduced `UnitTest-BuildPolicies` to validate that policy, initiative, and role definition builds match the expected output. (`.github/actions-pester/Test-BuildPolicies.Tests.ps1`)

### New Workflow for Policy Build Checks:
* Added `check-policy-build.yml` workflow to automatically check policy builds on pull requests. (`.github/workflows/check-policy-build.yml`)

### Removed Outdated Workflow:
* Removed the `update-portal.yml` workflow, which was responsible for updating the portal experience on pull requests. (`.github/workflows/update-portal.yml`)

### CODE REVIEW

**Comment by jtracey93** [comment]
AB#36739

File: .github/workflows/check-policy-build.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+          - "src/**.json"
+          - "src/**.bicep"
+    
+    env:
+      github_user_name: "github-actions"
+      github_email: "41898282+github-actions[bot]@users.noreply.github.com"
+      github_pr_number: ${{ github.event.number }}
+    
+    permissions:
+      contents: write
```
Comment (jtracey93) [inline .github/workflows/check-policy-build.yml:28]: These can be removed

File: docs/wiki/Whats-new.md [hunk 2]
```diff
@@ -47,6 +47,14 @@ This article will be updated as and when changes are made to the above and anyth
 
 Here's what's changed in Enterprise Scale/Azure Landing Zones:
 
+### August 2024
+
+> NOTE TO CONTRIBUTORS: Due to security compliance requirements, we've made core changes that mean we no longer automatically build the policies, initiatives and roles templates after changes in the `src` folder are committed. This means that you as a contributor must run the bicep build commands to generate the required outputs as part of your pull request. Depending on the files you've updated these are the commands (assuming you have bicep installed):
+> 
+> - `bicep build ./src/templates/policies.bicep --outfile ./eslzArm/managementGroupTemplates/policyDefinitions/policies.json`
+> - `bicep build ./src/templates/initiatives.bicep --outfile ./eslzArm/managementGroupTemplates/policyDefinitions/initiatives.json`
+> - `bicep build ./src/templates/roles.bicep --outfile ./eslzArm/managementGroupTemplates/roleDefinitions/customRoleDefinitions.json`
```
Comment (jtracey93) [inline docs/wiki/Whats-new.md:56]: can we also add this to the contribution guide please 👍

**Review by jtracey93 — CHANGES_REQUESTED** [review]

File: .github/workflows/check-policy-build.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            env:
+              GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    
+          # - name: Configure local git
+          #   run: |
+          #     echo "git user name  : $github_user_name"
+          #     git config --global user.name "$github_user_name"
+          #     echo "git user email : $github_email"
+          #     git config --global user.email "$github_email"
+
```
[the remark this answers is not in the document]
Comment (Springstone) [reply .github/workflows/check-policy-build.yml:59]: Oops, meant to but forgot.

**Review by jtracey93 — DISMISSED** [review]
LGTM

**Review by jtracey93 — APPROVED** [review]
LGTM

**Pull request merged**